go/coinstacks/osmosis/tx.go

```diff
@@ -29,13 +29,64 @@ func ParseMessages(msgs []sdk.Msg, events cosmos.EventsByMsgIndex) []cosmos.Mess
 			}
 			messages = append(messages, message)
 		case *gammtypes.MsgJoinPool:
-			message := cosmos.Message{
-				Addresses: []string{v.Sender},
-				From:      v.Sender,
-				Type:      v.Type(),
-				Value:     cosmos.CoinToValue(&v.TokenInMaxs[0]),
+			tokenInValues := cosmos.PoolEventAmountToValues(events["0"]["pool_joined"]["tokens_in"])
+			msgs := []cosmos.Message{
+				// token in 0
+				{
+					Addresses: []string{v.Sender},
+					Origin:    v.Sender,
+					From:      v.Sender,
+					Type:      v.Type(),
+					Value:     tokenInValues[0],
+				},
+				// token in 1
+				{
+					Addresses: []string{v.Sender},
+					Origin:    v.Sender,
+					From:      v.Sender,
+					Type:      v.Type(),
+					Value:     tokenInValues[1],
+				},
+				// token out (lp token)
+				{
+					Addresses: []string{v.Sender},
+					Origin:    v.Sender,
+					From:      events["0"]["transfer"]["sender"],
+					To:        v.Sender,
+					Type:      v.Type(),
+					Value:     cosmos.SerializedPoolTransferStringtoValue(events["0"]["transfer"]["amount"]),
```

Comment:
also prefer using message index from for loop to handle possible multiple join/exit messages in a single transaction
```suggestion
					Value:     cosmos.SerializedPoolTransferStringtoValue(events[strconv.Itoa(i)]["transfer"]["amount"]),
```